controller/commentsController.js

```diff
@@ -51,3 +51,53 @@ exports.flagComment = async (req, res) => {
     errHandler(error, res);
   }
 };
+
+exports.deleteComment = async (req, res) => {
+  const commentId = req.params.commentId;
+  const userEmail = req.body.commentOwnerEmail;
+  if (!commentId || !userEmail) {
+    return res.status(400).json({
+      data: null,
+      status: "Failed",
+      message: "Comment Id and email address required",
+    });
+  }
+  try {
+    const comment = await Comments.findOne({ _id: commentId }).populate(
+      "commentOwner"
+    );
+    if (!comment) {
+      return res.status(400).json({
+        data: null,
+        status: "Failed",
+        message: "Comment not found",
+      });
+    }
+    const email = await comment;
+    if (email.commentOwner.email.toLowerCase() == userEmail.toLowerCase()) {
+      const deleting = await Comments.findByIdAndDelete(commentId);
+      if (deleting) {
+        return res.status(200).json({
+          data,
+          status: "Success",
+          message: "Comment deleted successfully",
+        });
+      } else {
+        return res.status(400).json({
+          data: null,
+          status: "Failed",
+          message: "Cannot delete your comment at this time. Please try again",
+        });
+      }
+    } else {
+      return res.status(400).json({
+        data: null,
+        status: "Failed",
+        message:
+          "Comment cannot be deleted because you are not the owner of this comment.",
+      });
```

Comment:
```suggestion
      return next(CustomError(400, "Comment cannot be deleted because you are not the owner of this comment.");
```

Reply:
This is how you should implement the CustomError class. Please use this example to make the correct updates.